quil-py/quil/instructions/__init__.pyi

```diff
@@ -360,6 +425,124 @@ class BinaryLogic:
+    def __new__(cls, destination: MemoryReference, source: MemoryReference) -> Self: ...
+    @property
+    def destination(self) -> MemoryReference: ...
+    @destination.setter
+    def destination(self, destination: MemoryReference): ...
+    @property
+    def source(self) -> MemoryReference: ...
+    @source.setter
+    def source(self, source: MemoryReference): ...
+
+class Move:
+    def __new__(
+        cls, destination: MemoryReference, source: ArithmeticOperand
+    ) -> Self: ...
+    @property
+    def destination(self) -> MemoryReference: ...
+    @destination.setter
+    def destination(self, destination: MemoryReference): ...
+    @property
+    def source(self) -> ArithmeticOperand: ...
+    @source.setter
+    def source(self, source: ArithmeticOperand): ...
+
+class Exchange:
+    def __new__(cls, left: MemoryReference, right: MemoryReference) -> Self: ...
+    @property
+    def left(self) -> MemoryReference: ...
+    @left.setter
+    def left(self, left: MemoryReference): ...
+    @property
+    def right(self) -> MemoryReference: ...
+    @right.setter
+    def right(self, right: MemoryReference): ...
+
+@final
+class ComparisonOperand:
+    """
+    A Quil binary operand.
+
+    Variants:
+        ``literal_integer``: An integer literal.
+        ``literal_real``: An floating point literal.
+        ``memory_reference``: A Quil ``MemoryReference``.
+
+    Methods (for each variant):
+        ``is_*``: Returns ``True`` if the operand is that variant, ``False`` otherwise.
+        ``as_*``: Returns the inner data if it is the given variant, ``None`` otherwise.
+        ``to_*``: Returns the inner data if it is the given variant, raises ``ValueError`` otherwise.
+        ``from_*``: Creates a new ``BinaryOperand`` of the given variant from an instance of the inner type.
+    """
+
+    def inner(self) -> Union[int, float, MemoryReference]:
+        """
+        Returns the inner value of the variant.
+        """
+        ...
+    def is_literal_integer(self) -> bool: ...
+    def is_literal_real(self) -> bool: ...
+    def is_memory_reference(self) -> bool: ...
+    def as_literal_integer(self) -> Optional[int]: ...
+    def as_literal_real(self) -> Optional[float]: ...
+    def as_memory_reference(self) -> Optional[MemoryReference]: ...
+    def to_literal_integer(self) -> int: ...
+    def to_literal_real(self) -> float: ...
+    def to_memory_reference(self) -> MemoryReference: ...
+    @staticmethod
+    def from_literal_integer(literal_integer: int) -> "ComparisonOperand": ...
+    @staticmethod
+    def from_literal_real(literal_real: float) -> "ComparisonOperand": ...
+    @staticmethod
+    def from_memory_reference(
+        memory_reference: MemoryReference,
+    ) -> "ComparisonOperand": ...
+
+@final
+class ComparisonOperator(Enum):
+    Equal = "EQUAL"
+    GreaterThanOrEqual = "GREATERTHANOREQUAL"
+    GreaterThan = "GREATERTHAN"
+    LessThanOrEqual = "GREATERTHANOREQUAL"
+    LessThan = "LESSTHAN"
```

Comment:
Should these match the display values from the rust types? e.g, `"GE"` not `"GREATERTHANOREQUAL"`

https://github.com/rigetti/quil-rs/pull/196/files#diff-05aa3659ae4ba942340a9ef40aaf99f892949ce87121df395960513065495d36R218

Reply:
Definitely, fixed.